spring-cloud-gcp-storage/src/test/java/org/springframework/cloud/gcp/storage/GoogleStorageTests.java

```diff
@@ -260,6 +261,16 @@ public void testCreateRelativeSubdirs() throws IOException {
 		Assert.assertEquals("gs://test-spring/t1/r1/relative.png", relative.getURI().toString());
 	}
 
+	@Test
+	public void nullSignedUrlForNullBlob() throws IOException {
```

Comment:
I think we need more tests to fully cover the method.
1) `IOException` when `privateKeyPath` is empty.
2) Verify that `this.storage.signUrl` is called in the happy path.

Reply:
1. Will be not be nice to test, because env vars are not great to test with unit tests. Remember a few months ago we had issues with the CI stuff and tests that altered env vars. The GOOGLE_APPLICATION_CREDENTIALS env var is very widespread and set in stone, so I think this is the right way to access that private key JSON file. I think maybe we can just skip testing it. Even if this test would have failed, it's not a catastrophic or mysterious failure anyway. 

2. Will do.

Reply:
1. Why can't we rely on `spring.cloud.gcp.storage.credentials.location`, which is easy to mock in the test?

Reply:
Isn't that one only in the starter project? Also, other than a unit test of minimal (if any) value, what's the purpose? GOOGLE_APPLICATION_CREDENTIALS is much more widely known, and is mentioned directly in the documentation in this repo (not just the starter projects as the spring.cloud.gcp.xxxxx.credentials.location props).